snapcraft/internal/common.py

```diff
@@ -212,6 +212,16 @@ def _search_and_replace_contents(file_path, search_pattern, replacement):
             f.write(replaced)
 
 
+def get_terminal_width(max_width=MAX_CHARACTERS_WRAP):
+    if os.isatty(sys.stdout.fileno()):
+        width = shutil.get_terminal_size().columns
+    else:
+        width = max_width
+    if max_width:
+        width = min(max_width, width)
+    return width
```

Comment:
This seems wrong. 
You are calling: terminal_width = get_terminal_width(max_width=None). For not a tty, that will return None
Then you do: description_space = terminal_width - part_length - 5
So None - something - 5, will raise:
TypeError: unsupported operand type(s) for -: 'NoneType' and 'int'